### PR TITLE
Workaround to include relative readme into docs directory

### PR DESCRIPTION
**Pull request by abravalheri**
The workaround is described in https://github.com/readthedocs/recommonmark/issues/191

It is not perfect, but it does the job, and now readthedocs have something to show...

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -87,17 +87,25 @@
...
     app.add_config_value("recommonmark_config", params, True)
     app.add_transform(AutoStructify)
 
+    # from m2r to make `mdinclude` work
+    app.add_config_value('no_underscore_emphasis', False, 'env')
+    app.add_config_value('m2r_parse_relative_links', False, 'env')
+    app.add_config_value('m2r_anonymous_references', False, 'env')
+    app.add_config_value('m2r_disable_inline_math', False, 'env')
+    app.add_directive('mdinclude', MdInclude)
+
```
Comment (FlorianWilhelm) [inline docs/conf.py:108]: Do these changes also need to be added to the template itself? Meaning `AUTO_STRUCTIFY_CONF` in `extension.py`?

**Review by FlorianWilhelm — APPROVED** [review]
Nice workaround. Makes me think. On could also have a demo project for this extension. One could even have an RTD site for each demo project to showcase the default docs.

File: docs/conf.py (hunk 1 above)
Comment (abravalheri) [reply docs/conf.py:108]: We can add them to the template... it is a workaround and have some drawbacks... but they do allow usind `mdinclude`... which means that even files such as AUTHORS.md and CHANGELOG.md could be viable

Comment (abravalheri) [reply docs/conf.py:108]: There is a recent [PR](https://github.com/sphinx-doc/sphinx/pull/7739) that if accepted would lift the need for a workaround...

**Comment by abravalheri** [comment]
Hi @FlorianWilhelm I changed the implementation of the workaround...
The approach used by recommonmark's own documentation is to include a symbolic link file inside the `docs` directory, instead of trying to do a `include`.

References:
- https://github.com/readthedocs/recommonmark/issues/191
- https://github.com/sphinx-doc/sphinx/issues/701
- https://github.com/sphinx-doc/sphinx/pull/7739

I will try to also do this with the templates and generated project in another PR.

**Comment by FlorianWilhelm** [comment]
Okay, this even looks much cleaner. The only thing I find a bit dangerous is the symlinks. Do they even work on Windows? In https://github.com/sphinx-doc/sphinx/issues/701 they say it might be problematic. Once this is merged I need to check if the dsproject extension still works since it directly depends on this.

**Comment by abravalheri** [comment]
They "work" on Windows, if the user have setup the environment correctly.
Git-for-windows have some help about that in: https://github.com/git-for-windows/git/wiki/Symbolic-Links#creating-symbolic-links

I was thinking about catching the exception and then re-raising something like this:
```python
class SymlinkError(OSError):
    """\
    Impossible to create a symbolic link {{{link_path} => {original_file}}}.
    If you are using a non-POSIX operating system, please make sure that
    your user have the correct rights and that your system is corrctly
    configured.

    Please check the following references:
    http://github.com/git-for-windows/git/wiki/Symbolic-Links
    https://blogs.windows.com/windowsdeveloper/2016/12/02/ symlinks-windows-10/
    https://docs.microsoft.com/en-us/windows/win32/fileio/ creating-symbolic-links
    """
    def __init__(self, link_path, original_file, *args, **kwargs):
        msg = (self.__class__.__doc__ or "").format(original_file, link_path)
        super().__init__(msg, *args, **kwargs)
```

**Comment by FlorianWilhelm** [comment]
I think this would help Windows users a lot. A lot of Windows users struggle to set Git up, let alone something more complicated. We already had several issues on Windows where PyScaffold couldn't even find Git although it was supposed to be installed. Windows is just a pain... 
Nevertheless, I like your solution since it will work for the majority of users (which are Linux & MacOs anyway) without any trouble and it can work for Windows if everything is set up correctly.

**Comment by abravalheri** [comment]
I suppose we could go for it. If the Windows users still have problems even trying the documentation from git-for-windows and Microsoft, they still have 2 options: *WSL* and sticking with plain old rst... It is not like not using markdown would compromise a python project in terms of functionality...

**Pull request merged**